internal/services/codesigning/trusted_signing_certificate_profile_resource.go

```diff
@@ -0,0 +1,478 @@
+	IncludeCity             bool                                         `tfschema:"include_city"`
+	IncludeCountry          bool                                         `tfschema:"include_country"`
+	IncludePostalCode       bool                                         `tfschema:"include_postal_code"`
+	IncludeState            bool                                         `tfschema:"include_state"`
+	IncludeStreetAddress    bool                                         `tfschema:"include_street_address"`
+	ProfileType             certificateprofiles.ProfileType              `tfschema:"profile_type"`
+	Certificates            []CertificateModel                           `tfschema:"certificates"`
+	Status                  certificateprofiles.CertificateProfileStatus `tfschema:"status"`
+}
+
+type CertificateModel struct {
+	CreatedDate      string                                `tfschema:"created_date"`
+	EnhancedKeyUsage string                                `tfschema:"enhanced_key_usage"`
+	ExpiryDate       string                                `tfschema:"expiry_date"`
+	Revocation       []RevocationModel                     `tfschema:"revocation"`
+	SerialNumber     string                                `tfschema:"serial_number"`
+	Status           certificateprofiles.CertificateStatus `tfschema:"status"`
+	SubjectName      string                                `tfschema:"subject_name"`
+	Thumbprint       string                                `tfschema:"thumbprint"`
+}
+
+type RevocationModel struct {
+	EffectiveAt   string                               `tfschema:"effective_at"`
+	FailureReason string                               `tfschema:"failure_reason"`
+	Reason        string                               `tfschema:"reason"`
+	Remarks       string                               `tfschema:"remarks"`
+	RequestedAt   string                               `tfschema:"requested_at"`
+	Status        certificateprofiles.RevocationStatus `tfschema:"status"`
+}
+
+type TrustedSigningCertificateProfileResource struct{}
+
+var _ sdk.ResourceWithUpdate = TrustedSigningCertificateProfileResource{}
+
+func (r TrustedSigningCertificateProfileResource) ResourceType() string {
+	return "azurerm_trusted_signing_certificate_profile"
+}
+
+func (r TrustedSigningCertificateProfileResource) ModelObject() interface{} {
+	return &TrustedSigningCertificateProfileModel{}
+}
+
+func (r TrustedSigningCertificateProfileResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return certificateprofiles.ValidateCertificateProfileID
+}
+
+func (r TrustedSigningCertificateProfileResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(5, 100),
+				validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z][A-Za-z0-9]*(?:-[A-Za-z0-9]+)*$"),
```

Comment:
What is the meaning here to use the non-capture group (i.e. `(?:)`) since we are only matching? Or do you want the negative look ahead instead, which is not available in Go, iirc?

Reply:
it was converted from the regex on swagger: `^(?=.{5,100}$)[^0-9][A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`

Reply:
Looks like that's not correct based on the description.

Reply:
updated

Reply:
Still we need the `(?:)`?

Reply:
I think the answer is yes? to ensure there is a letter after `-`